src/Task/TalkNPC.pm

```diff
@@ -351,20 +559,44 @@ sub target {
 	return $self->{target};
 }
 
+#only for testing
+my $default_text = "eye lol";
+my $default_number = 1234;
+
 sub cancelTalk {
 	my ($self) = @_;
-	if ($ai_v{npc_talk}{talk} eq 'select') {
+	
+	if ($self->{error_message}) {
+		debug "Trying to auto close the conversation due to error.\n", 'ai_npcTalk';
+	}
+	
+	if ($ai_v{'npc_talk'}{'talk'} eq 'select') {
 		$messageSender->sendTalkResponse($self->{ID}, 255);
-	} elsif ($ai_v{npc_talk}{talk} ne 'close' && !$talk{canceled}) {
-		$messageSender->sendTalkCancel($self->{ID});
-		$talk{canceled} = 1;
+		$self->{sent_talk_response_cancel} = 1;
+		
+	} elsif ($ai_v{'npc_talk'}{'talk'} eq 'next') {
+		$messageSender->sendTalkContinue($talk{ID});
+		
+	} elsif ($ai_v{'npc_talk'}{'talk'} eq 'number') {
+		$messageSender->sendTalkNumber($talk{ID}, $default_number);
+		
+	} elsif ($ai_v{'npc_talk'}{'talk'} eq 'text') {
+		$messageSender->sendTalkText($talk{ID}, $default_text);
+		
+	} elsif ( $ai_v{'npc_talk'}{'talk'} eq 'buy_or_sell' ) {
+		$self->conversation_end;
+		$ai_v{'npc_talk'}{'talk'} = 'close';
+	
+	} elsif ( $ai_v{'npc_talk'}{'talk'} eq 'store' ) {
+		$self->conversation_end;
+		$ai_v{'npc_talk'}{'talk'} = 'close';
+		
+	} elsif ( $ai_v{'npc_talk'}{'talk'} eq 'sell' ) {
+		$self->conversation_end;
+		$ai_v{'npc_talk'}{'talk'} = 'close';
+		
```

Comment:
Well anything definitely would be improvement over DC'ing. Clearing the remaining steps sounds okay for me (after that either something would add more steps or NPC will timeout). We need input from more people on that.